picasso/src/main/java/com/squareup/picasso3/AssetRequestHandler.java

```diff
@@ -57,11 +56,16 @@ public void load(@NonNull Picasso picasso, @NonNull Request request, @NonNull Ca
     }
     boolean signaledCallback = false;
     try {
-      Source source = Okio.source(assetManager.open(getFilePath(request)));
+      BufferedSource source = Okio.buffer(Okio.source(assetManager.open(getFilePath(request))));
       try {
-        Bitmap bitmap = decodeStream(source, request);
+        ImageDecoder imageDecoder = request.decoderFactory.getImageDecoderForSource(source);
```

Comment:
This code is repeated in each RequestHandler and should probably be extracted into the abstract base class.